test-components/outbound-postgres/src/lib.rs

```diff
@@ -0,0 +1,121 @@
+use helper::{ensure, ensure_eq, ensure_matches, ensure_ok};
+
+use bindings::fermyon::spin2_0_0::{postgres, rdbms_types};
+
+helper::define_component!(Component);
+const DB_URL_ENV: &str = "DB_URL";
+
+impl Component {
+    fn main() -> Result<(), String> {
+        ensure_matches!(
+            postgres::Connection::open("hello"),
+            Err(postgres::Error::ConnectionFailed(_))
+        );
+        ensure_matches!(
+            postgres::Connection::open("localhost:10000"),
+            Err(postgres::Error::ConnectionFailed(_))
+        );
+
+        let address = ensure_ok!(std::env::var(DB_URL_ENV));
+        let conn = ensure_ok!(postgres::Connection::open(&address));
+        let rowset = ensure_ok!(test_numeric_types(&conn));
+        ensure!(rowset.rows.iter().all(|r| r.len() == 12));
+        ensure_matches!(rowset.rows[0][11], rdbms_types::DbValue::Floating64(f) if f == 1.0);
+
+        let rowset = ensure_ok!(test_character_types(&conn));
+        ensure!(rowset.rows.iter().all(|r| r.len() == 3));
+        ensure!(matches!(rowset.rows[0][0], rdbms_types::DbValue::Str(ref s) if s == "rvarchar"));
+
+        let pid1 = format!("{:?}", ensure_ok!(pg_backend_pid(&conn)));
+        let pid2 = format!("{:?}", ensure_ok!(pg_backend_pid(&conn)));
+        ensure_eq!(pid1, pid2);
+
+        Ok(())
```

Comment:
It looks like there are no tests around NULL handling.  We have had bugs around this in the past - it would be good to cover in the runtime tests, as it's hard to unit test robustly.